app/controllers/posts_controller.rb

```diff
@@ -19,9 +19,11 @@ def create
     respond_to do |format|
       format.html do
         if @new_post.save
-          redirect_to "/users/#{@new_post.author.id}/posts/", notice: 'Success!'
+          flash[:notice] = 'Post created successfully.'
+          redirect_to "/users/#{@new_post.author.id}/posts/"
         else
-          render :new, alert: 'Error occured!'
+          flash[:notice] = 'Post creation failed. Try again'
+          render :new
         end
       end
     end
```

Comment:
- [ ] As it is a [requirement](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/blog_app_validations_model_spec_n%2B1.md) for this project, `make sure that the N+1 problem is solved when fetching all posts and their comments for a user`, please kindly update your post#index action to handle this problem. You could handle it for example like so: `@posts = @user.posts.includes(:comments)` instead of `@posts = @user.posts`. Please kindly fix that.